features/admin.applications.v1/components/share-application-status-response-list.tsx

```diff
@@ -0,0 +1,360 @@
+/**
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import CircularProgress from "@oxygen-ui/react/CircularProgress";
+import { UIConstants } from "@wso2is/admin.core.v1/constants/ui-constants";
+import { IdentifiableComponentInterface } from "@wso2is/core/models";
+import {
+    DataTable,
+    ListLayout,
+    TableColumnInterface
+} from "@wso2is/react-components";
+import React, { ReactElement, ReactNode, useEffect, useState } from "react";
+import InfiniteScroll from "react-infinite-scroll-component";
+import {
+    Dropdown,
+    DropdownItemProps,
+    DropdownProps,
+    Grid,
+    Header,
+    Icon
+} from "semantic-ui-react";
+import useGetApplicationShareStatusUnits from "../api/use-get-application-share-status";
+import { ApplicationShareUnitStatus } from "../constants/application-management";
+import { ApplicationShareStatusUnitLinkInterface, ApplicationShareUnitStatusResponse,
+    OperationStatusSummary } from "../models/application";
+import "./share-application-status-response-list.scss";
+
+/**
+ * Proptypes for the share application status response list.
+ */
+interface ShareApplicationStatusResponseListProps extends IdentifiableComponentInterface {
+    operationId: string;
+    ["data-componentid"]: string;
+    operationSummary?: OperationStatusSummary;
+}
+
+/**
+ * Proptypes for the REST api params.
+ */
+interface Params {
+    shouldFetch?: boolean,
+    filter?: string;
+    limit?: number;
+    after?: string;
+    before?: string;
+}
+
+/**
+ * Application Share status page.
+ *
+ * @param props - Props injected to the component.
+ * @returns Application share status result.
+ */
+export const ShareApplicationStatusResponseList: React.FunctionComponent<ShareApplicationStatusResponseListProps> = (
+    props: ShareApplicationStatusResponseListProps
+): ReactElement => {
+
+    const {
+        operationId,
+        operationSummary,
+        ["data-componentid"]: componentId
+    } = props;
```

Comment:
Let's move prop desctructuring to the signature.

See https://github.com/wso2/identity-apps/blob/master/docs/write-code/CODE_QUALITY.md#refrain-from-using-defaultprops